File: 02_linked_lists/12_create_linked_list/create_linked_list.py
class Node:
  def __init__(self, val):
    self.val = val
    self.next = None

def create_linked_list(values):
  new_list = Node(None)
  tail = new_list

  for value in values:
    tail.next = Node(value)
    tail = tail.next

  return new_list.next

File: 02_linked_lists/12_create_linked_list/create_linked_list.js
class Node {
  constructor(val) {
    this.val = val;
    this.next = null;
  }
}

const createLinkedList = (values) => {
  const newList = new Node(null);
  let tail = newList

  for(let i = 0; i < values.length; i++) {
    let newNode = new Node(values[i])
    tail.next = newNode
    tail = tail.next
  }

  return newList.next
};

module.exports = {
  createLinkedList,
};

console.log(createLinkedList(["h", "e", "y"]));
// h -> e -> y
console.log(createLinkedList([1, 7, 1, 8]));
// 1 -> 7 -> 1 -> 8
console.log(createLinkedList(["a"]));
// a
console.log(createLinkedList([]));
// null